l10n_eu_nace/migrations/12.0.2.0.0/post-migration.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019 ACSONE SA/NV
+# Copyright 2020 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env):
+    nace_categories = env['ir.model_data'].search(
+        [
+            ('module', '=', 'l10_eu_nace'),
+            ('model', '=', 'res.partner.category'),
+        ]
+    )
+    for category_imd in nace_categories:
+        partners = env['res.partner'].search(
+            [('category_id', 'in', [category_imd.res_id])]
+        )
+
+        partners.write(
+            {
+                'nace_id': env.ref(
```

Comment:
Better to do this in SQL and in one shot (more or less this query - not exact):
```
UPDATE res_partner rp
SET nace_id = imd2.res_id
FROM ir_model_data imd1
LEFT JOIN ir_model_data imd2
    ON  imd2.module = imd1.module
    AND imd2.name = substring(imd1.name, 4)
WHERE imd1.name LIKE 'old_%'
    AND rp.category_id = imd1.res_is
```

Reply:
I did it as you suggested but couldn't test it as all our db are migrated. It will be nice if you can run it on existing data.